### ISSUE TITLE
Nodes showing as disconnected - websocket error

### ISSUE THREAD

**Issue by 877dev** [gablau/node-red-contrib-blynk-ws#20]
Hi @gablau hope you are well,

I had an issue today where node red was showing all Blynk nodes as disconnected, I rebooted node red with no effect, and updated to your latest version 1.0.3. 

Node red log:
```
13 Apr 14:15:23 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Start secure connection: wss://139.59.206.133:80/websockets
13 Apr 14:15:23 - [error] [blynk-ws-client:Blynk (LIVE SYSTEM)] Websocket Error: Client network socket disconnected before secure TLS connection was established
13 Apr 14:15:23 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Client Error
13 Apr 14:15:23 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Reconnect in 5 seconds...
13 Apr 14:15:23 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Websocket closed: wss://139.59.206.133:80/websockets
13 Apr 14:15:23 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Reconnect in 5 seconds...
```
I thought I had fixed this by going into node properties and changing ```wss://blynk-cloud.com/websockets``` to non-SSL ```ws://blynk-cloud.com/websockets```, deployed and it all connected fine. 

**If I try and change it back to SSL wss it disconnects all the nodes when I redeploy, now I am experimenting and cannot get it to connect at all!**

So I am asking:

1) Any idea what the issue is?

2) If the nodes disconnect again, is there a way to be notified/trigger a flow?

Many thanks
877

**Comment by 877dev** [comment]
After a while playing I have figured out how to replicate the issue.

I have two clients set up in node red, my 'LIVE' system and a 'read only' one for the wife.
This has been running fine like this for many months, both using the SLL wss webhook (cloud).

![image](https://user-images.githubusercontent.com/46575282/79131873-3758e200-7da1-11ea-8c1d-b830fc61a5d8.png)

If either of those is set to wss, it disconnects and stays disconnected. 
If both are set to ws (non-ssl) then is connects fine.

**Comment by gablau** [comment]
Hello,
this is a big problem, but I was unable to reproduce it locally.
However, I have noticed that there are some problems with opening and closing websockets, particularly when deploying a flow.

Could you try to make the settings that give error and then, after having done the deployment, restart the nodered service?
please attach me the logs you boot of nodered

let me know

**Comment by 877dev** [comment]
@gablau I am actually finding it hard to reproduce myself at the moment. It was happening earlier, now it's all working fine with wss on each server ( I added a third local server).

I will keep trying to reproduce.

Here is a snippet of the logs: 

`2020-10-01T11:20:49.331032535Z 1 Oct 12:20:49 - [info] [blynk-ws-client:Blynk LOCAL TEST] Start connection: ws://192.168.107.99:8080/websockets


2020-10-01T11:20:49.333951702Z 1 Oct 12:20:49 - [error] [blynk-ws-client:Blynk LOCAL TEST] Websocket Error: connect ECONNREFUSED 192.168.107.99:8080


2020-10-01T11:20:49.334686026Z 1 Oct 12:20:49 - [info] [blynk-ws-client:Blynk LOCAL TEST] Client Error


2020-10-01T11:20:49.334771618Z 1 Oct 12:20:49 - [info] [blynk-ws-client:Blynk LOCAL TEST] Reconnect in 5 seconds...


2020-10-01T11:20:49.334823247Z 1 Oct 12:20:49 - [info] [blynk-ws-client:Blynk LOCAL TEST] Websocket closed: ws://192.168.107.99:8080/websockets


2020-10-01T11:20:49.334872098Z 1 Oct 12:20:49 - [info] [blynk-ws-client:Blynk LOCAL TEST] Reconnect in 5 seconds...


2020-10-01T11:20:49.665206090Z 1 Oct 12:20:49 - [info] [blynk-ws-client:Blynk LOCAL TEST] Start secure connection: wss://192.168.107.99:9443/websockets


2020-10-01T11:20:49.793293111Z 1 Oct 12:20:49 - [info] [blynk-ws-client:Blynk LOCAL TEST] Connect failed. code: 9


2020-10-01T11:20:54.335042373Z 1 Oct 12:20:54 - [info] [blynk-ws-client:Blynk LOCAL TEST] Start connection: ws://192.168.107.99:8080/websockets


2020-10-01T11:20:54.339077952Z 1 Oct 12:20:54 - [error] [blynk-ws-client:Blynk LOCAL TEST] Websocket Error: connect ECONNREFUSED 192.168.107.99:8080


2020-10-01T11:20:54.339763055Z 1 Oct 12:20:54 - [info] [blynk-ws-client:Blynk LOCAL TEST] Client Error


2020-10-01T11:20:54.339918886Z 1 Oct 12:20:54 - [info] [blynk-ws-client:Blynk LOCAL TEST] Reconnect in 5 seconds...


2020-10-01T11:20:54.340429972Z 1 Oct 12:20:54 - [info] [blynk-ws-client:Blynk LOCAL TEST] Websocket closed: ws://192.168.107.99:8080/websockets


2020-10-01T11:20:54.340527712Z 1 Oct 12:20:54 - [info] [blynk-ws-client:Blynk LOCAL TEST] Reconnect in 5 seconds...`

**Comment by 877dev** [comment]
@gablau ok it happened again this morning.

![image](https://user-images.githubusercontent.com/46575282/94904540-62400b00-0493-11eb-8afd-5f11cfaa1bbb.png)
In this case, I have a custom server running in a Docker container. The top node was working fine all of yesterday, the server was set to wss 9443 connection.

This morning I added the bottom node, and edited the server connection to ws port 8080. Then when redeployed it shows the error.

I immediately restarted node red via Portainer and here is the log file (I deleted some Alexa info but otherwise complete):
https://gist.github.com/877dev/1d94e480ec083274c1ad8deba48ccc33

**Comment by 877dev** [comment]
Feb 2021, the problem has reoccurred over this last week. 
I restarted node red, here's the logs (I deleted some sensitive Alexa info only):
[_nodered_logs.log](https://github.com/gablau/node-red-contrib-blynk-ws/files/6018131/_nodered_logs.log)

**Comment by 877dev** [comment]
Darn, just changed a few unrelated flows and the error is back, can't seem to fix it this time.....

```
21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Start secure connection: wss://139.59.206.133:80/websockets


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] LOG PINS []


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Start secure connection: wss://blynk-cloud.com/websockets


21 Feb 20:37:56 - [info] [blynk-ws-client:Family Blynk account] LOG PINS []


21 Feb 20:37:56 - [info] [blynk-ws-client:Family Blynk account] Start secure connection: wss://blynk-cloud.com/websocket


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk TEST LOCAL] LOG PINS []


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register LCD node pin: 70


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 0


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 104


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: style-btn pin: 104


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: app


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 106


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 108


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: style-btn pin: 106


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: style-btn pin: 108


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: app


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: app


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 97


21 Feb 20:37:56 - [info] [blynk-ws-client:Family Blynk account] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 62


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 3


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 18


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 105


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 116


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 117


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 102


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: style-btn pin: 102


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: app


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 100


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: style-btn pin: 100


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: app


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 103


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 109


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 111


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 110


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 13


21 Feb 20:37:56 - [info] [blynk-ws-client:Family Blynk account] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 50


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 49


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 10


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 4


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 7


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 101


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 254


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 61


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 62


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register input node - type: write pin: 63


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Family Blynk account] Register notify node


21 Feb 20:37:56 - [info] [blynk-ws-client:Family Blynk account] Register notify node


21 Feb 20:37:56 - [error] [blynk-ws-client:Blynk (LIVE SYSTEM)] Websocket Error: Client network socket disconnected before secure TLS connection was established


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Client Error


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Reconnect in 5 seconds...


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Websocket closed: wss://139.59.206.133:80/websockets


21 Feb 20:37:56 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Reconnect in 5 seconds...


21 Feb 20:37:58 - [warn] [blynk-ws-client:Blynk (LIVE SYSTEM)]   wss.//139.59.206.133.80/websockets


21 Feb 20:37:58 - [info] [blynk-ws-client:Blynk TEST LOCAL] Connection disabled by configuration


21 Feb 20:38:01 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Start secure connection: wss://139.59.206.133:80/websockets


21 Feb 20:38:01 - [error] [blynk-ws-client:Blynk (LIVE SYSTEM)] Websocket Error: Client network socket disconnected before secure TLS connection was established


21 Feb 20:38:01 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Client Error


21 Feb 20:38:01 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Reconnect in 5 seconds...


21 Feb 20:38:01 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Websocket closed: wss://139.59.206.133:80/websockets


21 Feb 20:38:01 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Reconnect in 5 seconds...


21 Feb 20:38:02 - [error] [blynk-ws-client:Family Blynk account] Websocket Error: socket hang up


21 Feb 20:38:02 - [info] [blynk-ws-client:Family Blynk account] Client Error


21 Feb 20:38:02 - [info] [blynk-ws-client:Family Blynk account] Reconnect in 5 seconds...


21 Feb 20:38:02 - [info] [blynk-ws-client:Family Blynk account] Websocket closed: wss://blynk-cloud.com/websocket


21 Feb 20:38:02 - [info] [blynk-ws-client:Family Blynk account] Reconnect in 5 seconds...


21 Feb 20:38:06 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Start secure connection: wss://139.59.206.133:80/websockets


21 Feb 20:38:06 - [error] [blynk-ws-client:Blynk (LIVE SYSTEM)] Websocket Error: Client network socket disconnected before secure TLS connection was established


21 Feb 20:38:06 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Client Error


21 Feb 20:38:06 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Reconnect in 5 seconds...


21 Feb 20:38:06 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Websocket closed: wss://139.59.206.133:80/websockets


21 Feb 20:38:06 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Reconnect in 5 seconds...


21 Feb 20:38:07 - [info] [blynk-ws-client:Family Blynk account] Start secure connection: wss://blynk-cloud.com/websocket


21 Feb 20:38:07 - [error] [blynk-ws-client:Family Blynk account] Websocket Error: socket hang up


21 Feb 20:38:07 - [info] [blynk-ws-client:Family Blynk account] Client Error


21 Feb 20:38:07 - [info] [blynk-ws-client:Family Blynk account] Reconnect in 5 seconds...


21 Feb 20:38:07 - [info] [blynk-ws-client:Family Blynk account] Websocket closed: wss://blynk-cloud.com/websocket


21 Feb 20:38:07 - [info] [blynk-ws-client:Family Blynk account] Reconnect in 5 seconds...


21 Feb 20:38:08 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Websocket closed: wss://139.59.206.133:80/websockets


21 Feb 20:38:08 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Reconnect in 5 seconds...


21 Feb 20:38:11 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Start secure connection: wss://139.59.206.133:80/websockets


21 Feb 20:38:11 - [error] [blynk-ws-client:Blynk (LIVE SYSTEM)] Websocket Error: Client network socket disconnected before secure TLS connection was established


21 Feb 20:38:11 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Client Error


21 Feb 20:38:11 - [info] [blynk-ws-client:Blynk (LIVE SYSTEM)] Reconnect in 5 seconds...
```

**Comment by Peterkn2001** [comment]
Are you sure you changed the url for your “ LIVE SYSTEM” connection?
Did you restart Node-Red afterwards?

Pete.

**Comment by 877dev** [comment]
Hi Pete,

I did not need to change the LIVE url at first, it was working fine. Restarting seemed to give mixed results, sometimes connecting or not.

Last night I changed both to "wss://blynk-cloud.com/websocket" and restarted, and it seems to be fine - (again). Fingers crossed...

**Comment by 877dev** [comment]
@Peterkn2001 for no apparent reason I am today getting this every so often, nodes are connecting though...
Are you having the same?

![image](https://user-images.githubusercontent.com/46575282/109344041-19d02200-7866-11eb-914e-2988556006d4.png)

**Comment by gablau** [comment]
Hi @877dev,
i released a new version [1.0.6](https://github.com/gablau/node-red-contrib-blynk-ws/releases/tag/1.0.6), could you see if it solves this problem?

I fixed some things about the websocket connection.

Best Regards

**Comment by 877dev** [comment]
Hi @gablau,

I just updated and it seems to be working fine so far, although I have had no issues for a while now.
Thanks
